src/GenesisApp.cpp

```diff
@@ -41,13 +43,124 @@ GenesisApp::GenesisApp()
 ////////////////////////////////////////////////////////////////////////
 	: BApplication(GENESIS_APP_SIG)
 {
+	fMainWindowsList = new BObjectList<GenesisWindow>(5);
+	fQuitConfirmed = false;
+	fPrefWindow = NULL;
+}
+
+////////////////////////////////////////////////////////////////////////
+void GenesisApp::NewWindow()
+////////////////////////////////////////////////////////////////////////
+{
+	GenesisWindow* w;
 
+	w = new GenesisWindow();
+	w->Show();
+
+	fMainWindowsList->AddItem(w);
+}
+
+////////////////////////////////////////////////////////////////////////
+void GenesisApp::OpenPreferences(GenesisWindow *w = NULL)
+////////////////////////////////////////////////////////////////////////
+{
+	if (fPrefWindow == NULL)
+	{
+		fPrefWindow = new GenesisPreferencesWindow();
+		fPrefWindow->Show();
+	}
+	fPrefWindow->Activate();
+
+	if (w != NULL)
+	{
+		fPrefWindow->CenterIn(w->Frame());
```

Comment:
It's a bit weird to see the already opened preferences window jump into the center of the main window. What do you think about giving the main window's coordinates as parameter when creating the preferences window and center the prefs window in those coordinates in the pref window c'tor.
The prefs window will then stick to where the user may have dragged it.